pico/configdriven/services/src/main/java/io/helidon/pico/configdriven/services/ConfigBeanRegistry.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.configdriven.services;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import io.helidon.builder.config.spi.BasicConfigBeanRegistry;
+import io.helidon.builder.config.spi.ConfigBeanInfo;
+
+/**
+ * The highest ranked/weighted implementor of this contract is responsible for managing the set of
+ * {@link io.helidon.builder.config.ConfigBean}'s that are active, along with whether the application is configured to
+ * support dynamic aspects (i.e., dynamic in content, dynamic in lifecycle, etc.).
+ */
+public interface ConfigBeanRegistry extends BasicConfigBeanRegistry {
+
+    /**
+     * The config bean registry is initialized as part of Pico's initialization, which happens when the service registry
+     * is initialized and bound.
+     *
+     * @return true if the config bean registry has been initialized
+     */
+    boolean ready();
+
+    /**
+     * These are the services that are configurable, mapping to the configuration beans each expects.
+     * Each entry in the returned map is the master/root for the config beans it manages. The result, therefore, is
+     * not associated with config beans. Use {@link #configuredServiceProviders()} for configured service instances.
+     *
+     * @param <T>  the service type
+     * @param <CB> the config bean type
```

Comment:
Generic does not make sense here (same for all methods on this interface). The method does not have the types specified in its parameters, so it cannot return the correct type (it cannot infer the types from anything)
This is a bug, as I can write
`Map<ConfiguredServiceProvider<java.util.Time, java.util.Time>> = configurableServiceProviders();` 
Which is a valid signature, but the method will most likely return something different, and type safety is gone...

Reply:
pushed

Reply:
`configBeansByConfigKey` and `configBeanMapByConfigKey` are still wrong

Reply:
Also both methods have an optional parameter, which is not correct